Common/include/geometry/CGeometry.hpp

```diff
@@ -910,6 +911,12 @@ class CGeometry {
    */
   inline virtual void SetRestricted_GridVelocity(const CGeometry *fine_grid) {}
 
+  /*!
+   * \brief Compute the surface area of all global markers.
+   * \param[in] config - Definition of the particular problem.
+   */
+  void ComputeSurfaceArea(const CConfig *config);
```

Comment:
Call it in CGeometry::UpdateGeometry or the sensitivities will be incorrect.
Might also be needed for the OF in multizone adjoints?

Reply:
added the call in updateGeometry in 288a2eb . `UpdateGeometry` is only ever called for deforming meshes. When I moved the ComputeSurfaceArea routine only to `UpdateGeometry` my case segfaulted because ComputeSurfaceArea was never called and the vector never resized and filled. The call to `ComputeSurfaceArea` coming from the CDriver ctor (via `Geometrical_Preprocessing`) is always evaluated at startup (except for DG-FEM cases)

Reply:
I am unsure where you think this would go wrong for multizone adjoints :thinking:

Reply:
Yes I said to call it, not to move it :wink:
Just thinking of the places where `Set_Heatflux_Areas` was used.

Reply:
I wanted to tackle `Set_Heatflux_Areas` at another point anyway as it is somewhat connected to `Heat_Fluxes` which computes OF's and hist-output for the HeatSolver. And in there I dont like 2 things:
1. On isothermal walls, the cfg T is used for HF computation. The Isothermal wall is weakly enforced so this could create quite a different value compared to a purely postprocessed (eg via paraview) result
2. AvgTemp is only reported for Heatflux-walls (not even for CHT-interfaces). Imo it should be users choice where he wants which quantities even if they are trivial

I reverted the cut-down of `Set_Heatflux_Areas` here e198acf  because I would need to care about above points and that might be better done in a separate PR